### DISCUSSION TITLE
Fire and forget for long running tasks are now supported??

### DISCUSSION THREAD

**Discussion by nv-cho** [vercel/next.js#62705] [Help] [answered]
### Summary

Hi! I am using Next.js 13.5.2 with the page router and its API routes as my backend.

I have the following flow consisting of three endpoints A, B and C. B and C are endpoints dedicated to long-running tasks.

A is called from the client, then A calls B but does not wait for the response (the UI does not care about the immediate response), finally A returns a status 200 to the client.

B calls C also without waiting for a response. This flow only worked sometimes (most of the time it failed), I assumed it was because fire and forget calls were not supported. It is worth mentioning that both calls to B and C have a catch block, and this is what I get in Sentry.

For endpoint B: 

connect ETIMEDOUT 76.76.21.22.22:443

For endpoint C:

write EPROTO C00F7099E07F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:ssl/record/ssl3_record.c:355:

But since two days ago I think, this flow is always working.

What I want to know is.

1. was my original assumption correct?
2. has something changed, why does it work now?

Please! someone can give me some clarity? this is making me crazy haha

**Comment by icyJoseph** [comment]
Hi, 

Assuming you had deployed to Vercel, the problem with endpoint B might just be that, the serverless task A, was terminated, and so the B endpoint kills the connection. Not sure about C, that sounds more like TLS configuration.

**Reply by nv-cho** [reply]
yes, it's deployed on Vercel

**Comment by AlbertMarashi** [comment, marked as answer]
Update for anyone coming from google:

- https://vercel.com/changelog/waituntil-is-now-available-for-vercel-functions

```sh
npm i @vercel/functions
```

```ts
import { waitUntil } from "@vercel/functions";

// within the api handler
// will continue running even after the response as been returned
waitUntil(new Promise(resolve => setTimeout(resolve(), 500)));

// immediately return your response
return response
```

@nv-cho please mark this as the new accepted answer so everyone can see it.

**Reply by nv-cho** [reply]
thanks!
